v1/lists.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+const P = require('bluebird');
+const preq = require('preq');
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const spec = HyperSwitch.utils.loadSpec(`${__dirname}/lists.yaml`);
+
+class ReadingLists {
+    /**
+     * @param {!Object} options RESTBase options object.
+     *   Apart from standard options, the following are recognized:
+     *   - summaryConcurrency: the max number of parallel requests to summary endpoint (default: 10)
+     */
+    constructor(options) {
+        this.options = options;
+    }
+
+    /**
+     * Transform the continuation data into a string so it is easier for clients to deal with.
+     * @param {!Object|undefined} continuation Continuation object returned by the MediaWiki API.
+     * @return {!String|undefined} Continuation string.
+     */
+    flattenContinuation(continuation) {
+        return JSON.stringify(continuation);
+    }
+
+    /**
+     * Inverse of flattenContinuation.
+     * @param {!String} continuation Continuation string returned by flattenContinuation().
+     * @return {!Object|undefined} Continuation object.
+     */
+    unflattenContinuation(continuation) {
+        const sanitizedContinuation = {};
+        if (typeof continuation === 'string') {
+            try {
+                continuation = JSON.parse(continuation);
+            } catch (e) {
+                this.options.log('error/unflatten', {
+                    msg: e.stack,
+                    json: continuation,
+                });
+                throw e;
+            }
+            // Make sure nothing malicious can be done by splicing the continuation data
+            // into the API parameters.
+            const allowedKeys = ['continue', 'rlcontinue', 'rlecontinue'];
+            for (const key of allowedKeys) {
+                if (typeof continuation[key] !== 'object') {
+                    sanitizedContinuation[key] = continuation[key];
+                }
+            }
+        }
+        return sanitizedContinuation;
+    }
+
+    /**
+     * Convert an array of values into the format expected by the MediaWiki API.
+     * @param {!Array} list A list containing strings and numbers.
+     * @return {!String}
+     */
+    flattenMultivalue(list) {
+        return list.join('|');
+    }
+
+    /**
+     * Handle the /list/{id}/entries endpoint (get entries of a list).
+     * @param {!HyperSwitch} hyper
+     * @param {!Object} req The request object as provided by HyperSwitch.
+     * @return {!Promise<Object>} A response promise.
+     */
+    getListEntries(hyper, req) {
+        return hyper.post({
+            uri: new URI([req.params.domain, 'sys', 'action', 'rawquery']),
```

Comment:
There is actually no need for `rawquery` here, you can use `query` since the response contains the `query` field, and hence will not fail.

Reply:
`query` throws an error if there is no `query.pages` field in the response. Reading lists don't deal with pages (as understood by the MW API which has no concept of remote pages) so that wouldn't work here.